lib/alchemy/config_loader.rb

```diff
@@ -0,0 +1,69 @@
+module Alchemy
+  class ConfigLoader
+    attr_reader :which
+
+    def initialize(which)
+      @which = which.sub(/\.yml$/, '')
+    end
+
+    # Reads the config file named +<which>.yml+ from +config/alchemy/+ folder.
+    #
+    def load_all
+      if paths.empty?
+        raise LoadError, "Could not find #{which}.yml file! Please run `rails generate alchemy:scaffold`"
+      else
+        # OPTIMIZE: remove duplicates?
+        paths.map(&method(:load_file)).reduce(&method(:merge))
```

Comment:
Imo map-reduce is the most explicit and elegant way here because its a common pattern, not only in ruby. I also think rewriting this makes it more complex. If one doesn't understand the implementation its really easy to find out because there is a lot of documentation about map & reduce on the internet. Also by reading the method names `load_files` and `merge` that are used here makes the behaviour quite obvious imo.

If you want to make it accessible for people who don't understand what happens there lets just add a comment that gives an explanation in simple english: `# Load all files and merge them into one`. That is enough to know here and I believe most people can even understand that by only reading that line of code.